ai/src/ai/common/prompt_fragment.py

```diff
@@ -12,6 +13,11 @@ class PromptFragment(BaseModel):
   role: Literal["user", "assistant", "system"]
   chain_of_thought: bool = False
 
+  @field_validator("id", mode="after")
```

Comment:
optional: we could have a parent class like BaseEntity similar to BaseExample https://github.com/google/mesop/blob/525ccd8e2c8dc8bd9095a36407754452312b200b/ai/src/ai/common/example.py#L54 to consolidate the validation logic

Reply:
also, it'll make it easier to add timestamps #938

Reply:
Good idea. Updated.